content/blog/2022-05-23-vectorization/index.md

```diff
@@ -0,0 +1,238 @@
++++
+title = "Vectorization for Bril"
+[[extra.authors]]
+name = "Jonathan Tran"
++++
+
+## Introduction
+
+In this project, I implemented auto-vectorization in the Bril language. I added a basic vector instruction set to the Bril language and implemented the instruction set with an extension of the reference interpreter, called `brili-vc`. I then implemented various approaches towards vectorization, beginning with a naive form of vectorization which creates vector instructions whenever possible. I also implemented a form of vectorization based on local value numbering, which opportunistically attempts to reuse vectors as much as possible. As a goal for this project, I wanted to develop correct vectorization algorithms, and measure how many vector instructions were created. To measure correctness, I created my own test cases, and to measure how many vector instructions were created, I used the CS 6120 Bril Benchmarks and calculated the number of vector instructions that were generated by instrumenting the `brili-vc` interpreter to track relevant statistics. My implementation is at the links: [vectorization](https://github.com/JonathanDLTran/CS6120-Jonathan) and [brili-vc](https://github.com/JonathanDLTran/bril/tree/main/bril-ts).
+
+## Vector Language
+
+The design of the vector language was based off resources from several students who had previously taken CS 6120, as well as the LLVM vector language. The students' languages are located at [vril](https://www.cs.cornell.edu/courses/cs6120/2019fa/blog/vril-vector-bril/), and [vector-instruction-support](https://www.cs.cornell.edu/courses/cs6120/2019fa/blog/interpreter-vector-support/). The LLVM vector language can be found be found at [llvm-vector](https://llvm.org/docs/LangRef.html#vector-operations).
```

Comment:
FWIW, this paragraph might be a tiny bit more readable if you used the names of the authors rather than the URL slugs for the blog posts?

Reply:
I fixed this by using the authors' names.